### PR TITLE
Adding signal callback to default gui input types

### PR DESCRIPTION
**Pull request by christinoleo**
I was poking around while checking if this library would be good for my needs and wanted to contribute back.

I noticed that in Vector3 Math, one of the possible outputs is a scalar, while the other ones are Vector3. I did some changes to allow the output type to change based on a signal connection, though I am not sure if it matches the overall architecture used in the package. I believe that with this, it would be easier to make more generic nodes where inputs and outputs can change depending on some of the parameters, but again I don't know if it matches with your goals.

Let me know if it is best to do some other kind of change.

### CODE REVIEW

**Comment by HungryProton** [comment]
That's great! I think we could just make this a default, add an empty method "_on_default_gui_dropdown" on the concept graph itself and always make the connection.
I'm working on a way to disable an input and this should be complete

File: src/nodes/maths/vector3_math.gd [hunk 1]
```diff
@@ -45,3 +46,26 @@ func _generate_outputs() -> void:
...
+		"Substract": 
+			set_output(0, "", ConceptGraphDataType.VECTOR3)
+		"Cross product": 
+			set_output(0, "", ConceptGraphDataType.VECTOR3)
+		"Dot product": 
+			set_output(0, "", ConceptGraphDataType.SCALAR)
+		"Bounce": 
+			set_output(0, "", ConceptGraphDataType.VECTOR3)
+		"Normalize": 
+			set_output(0, "", ConceptGraphDataType.VECTOR3)
```
Comment (HungryProton) [inline src/nodes/maths/vector3_math.gd:65]: Here for example, the second vector input is not used so you can call remove_input(2) to only leave one

But that also means we need to call  set_input(2, "B", ConceptGraphDataType.VECTOR3, opts) everywhere two inputs are needed

**Comment by HungryProton** [comment]
I changed a few things regarding how the dropdown was generated and merged the PR, thank you for your contribution!

**Comment by christinoleo** [comment]
Thanks for letting me help out!
Your update on this PR also made me learn some new things about GDScript too, which is also pretty nice.

**Comment by HungryProton** [comment]
No problem! Sorry for merging this without waiting for you, I broke a lot of things because of other issues unrelated to this PR so I had no way to explain why some things didn't worked anymore.
Anyway, your code is now in the master branch as part of the 0.3 release :D

**Pull request merged**